artemis-core-client/src/main/java/org/apache/activemq/artemis/core/protocol/core/impl/ChannelImpl.java

```diff
@@ -698,7 +698,7 @@ public void transferConnection(final CoreRemotingConnection newConnection) {
 
          if (logger.isTraceEnabled()) {
             logger.trace("RemotingConnectionID={} transferConnection to new RemotingConnectionID={}",
-                         (connection == null ? "NULL" : connection.getID()),
+                         connection.getID(),
```

Comment:
It would make sense to remove the check on the related line immediately following this too, especially as _newConnection_ replaces _connection_, but would be NPEd on in the next bit of code after this logging call before it can happen. Perhaps could also add a newConnection null check at the start of the method to make it more explicit?

Reply:
Done. I think. Take a look at the recent push.